opensubdiv: Get building for windows

expenses opened the pull request:
## Description of changes

- opensubdiv requires the python interpreter in order to generate a few things, but no python libraries so this can be a native dependency.
- It's possible to build ocl-icd for windows (with additional patches: https://github.com/NixOS/nixpkgs/pull/290100) but opensubdiv doesn't strictly require opencl (or any GPU compute platform) so I think it's acceptable to skip that for the time being.
- Because _only_ static libraries are built, `moveToOutput` removes everything from `$out` and breaks the build (similar to https://github.com/NixOS/nixpkgs/pull/289217). I took the approach of symlinking `$out` and `$static` but there might be a better way to do this. 

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

expenses committed:
opensubdiv: Get building on windows

Signed-off-by: John Titor <50095635+JohnRTitor@users.noreply.github.com>

Mindavi commented on pkgs/development/libraries/opensubdiv/default.nix:
```diff
@@ -58,10 +60,11 @@ stdenv.mkDerivation rec {
       (lib.cmakeBool "NO_OPENCL" (!openclSupport))
       (lib.cmakeBool "NO_CUDA" (!cudaSupport))
-    ] ++ lib.optionals (!stdenv.isDarwin) [
+    ] ++ lib.optionals stdenv.isLinux [
```
Maybe the previous person meant `isUnix && ! isDarwin` here.

expenses replied:
I'm happy to make that change, but I'm not sure if it will make things build on non-linux (& non-darwin) unixes.

expenses replied:
Done

JohnRTitor reviewed (commented):
Rebased 24b3f4c0f01d4b4867efbb0796b30e2b4ddb942c.

JohnRTitor committed:
opensubdiv: move python3 to nativeBuildInputs

opensubdiv requires the python interpreter in order to generate a few things, but no python libraries so this can be a native dependency

Signed-off-by: John Titor <50095635+JohnRTitor@users.noreply.github.com>

JohnRTitor committed:
opensubdiv: modernize package

rec -> finalAttrs
sha256 -> hash
rev -> tag

Signed-off-by: John Titor <50095635+JohnRTitor@users.noreply.github.com>
